Cleanup all os_keystone_* modules

SamYaple opened the pull request:
I have attempted to match the modules to both existing os_\* modules
and each other to provide some consistency and flow. This includes
structure and docs.

These changes have all been tested, except for the domain modules
since shade doesn't have support for those functions yet.

SamYaple committed:
Cleanup all os_keystone_* modules

I have attempted to match the modules to both existing os_* modules
and each other to provide some consistency and flow. This includes
structure and docs.

These changes have all been tested, except for the domain modules
since shade doesn't have support for those functions yet.

Shrews commented on cloud/openstack/os_keystone_domain.py:
```diff
@@ -60,60 +59,86 @@
+                # domains did not match
+                cloud.update_domain(domain_kwargs)
+                module.exit_json(changed=True,
+                                 result='updated',
```
Remove 'result'. We are no longer propagating its use. It's redundant.

SamYaple replied:
Ah thanks. I really don't like having result in there.

Though I can see it being useful because there are some returns that you can't determine what happened with it. That being said it is not normal ansible an I am happy to remove it

Shrews commented on cloud/openstack/os_keystone_domain.py:
```diff
@@ -60,60 +59,86 @@
-            _get_server_state(module, cloud)
-            _create_server(module, cloud)
+                module.exit_json(changed=False,
+                                 result='success',
```
Remove 'result'.

Shrews commented on cloud/openstack/os_keystone_domain.py:
```diff
@@ -60,60 +59,86 @@
-            _delete_server(module, cloud)
+            if domain is not None:
+                cloud.delete_domain(name_or_id=domain.id)
+                module.exit_json(changed=True, result='deleted')
```
Remove 'result'.

Shrews commented on cloud/openstack/os_keystone_domain.py:
```diff
@@ -60,60 +59,86 @@
+                cloud.delete_domain(name_or_id=domain.id)
+                module.exit_json(changed=True, result='deleted')
+            else:
+                module.exit_json(changed=False, result='success')
```
Remove 'result'.

Shrews commented on cloud/openstack/os_keystone_service.py:
```diff
@@ -51,32 +50,43 @@
-           service_a['description'] == service_b['description']
+    return service_a.name == service_b['name'] and \
+           service_a.type == service_b['service_type'] and \
+           service_a.description == service_b['description']
```
We should not be using object attribute notation. Assume a dict is returned.

SamYaple replied:
My bad, this is a change that slipped through when I was messing around with the code. Will fix!

SamYaple committed:
Removed 'result' returns as they are redundant

Reverted bad change with how a Bunch was addressed (as an object
rather than as a dict).
